scripts/irods/test/test_misc.py

```diff
@@ -196,3 +196,37 @@ def test_server_removes_leftover_temporary_rulebase_pid_files_on_startup__issue_
             for fn in [pid_file_1, pid_file_2, pid_file_3]:
                 self.assertFalse(os.path.exists(fn))
 
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "skip for topology testing")
+    def test_server_respawns_processes__issue_4977(self):
+        import psutil
+        import time
+        from collections import defaultdict
+        main_server = None
+        # Find the main server process
+        for proc in psutil.process_iter():
+            # It won't have any parents.
+            if proc.name() == 'irodsServer' \
+               and proc.parent().name() != 'irodsServer':
+                main_server = proc
+                # Kill the agent factory and the rule server
+                for child in proc.children(recursive=True):
+                    child.kill()
+                break
+        #It should run every 5 seconds but it's probably best to wait a lot longer just in case
+        time.sleep(15)
+
+        self.assertTrue(len(main_server.children()) >= 2)
+        # If the API is up after this it seems it should be entirely working.
+        self.admin.assert_icommand(['itouch', 'hello'])
+        self.admin.assert_icommand(['irm','hello'])
+        seen_process_names = defaultdict(int)
+        for i in main_server.children():
+            seen_process_names[i.name()] += 1
+        # There should only be a single delay server instance
+        self.assertTrue(seen_process_names['irodsReServer'] == 1)
+        # Test the delay server
+        self.admin.assert_icommand(['irule', '-r',
+                                    "irods_rule_engine_plugin-irods_rule_language-instance",
+                                    "'delay(\"\") { writeLine(\"serverLog\", \"DID IT!\"); }'",
+                                    'null',
+                                    'ruleExecOut'])
```

Comment:
rather than writing to the log... we could add an AVU to something, sleep 31s, and then confirm that the AVU is in the system.   no log checking.

Reply:
Alright, I'm gonna need some guidence on how to make that work.

Reply:
`msiModAVUMetadata` is the thing in the rule itself to set an AVU... and then an assert in python with `imeta ls -d` to confirm it made it in after the delay server fires the rule.

https://docs.irods.org/4.2.9/doxygen/extractAvuMS_8cpp.html#a547b0640dacd3b8d58979688f6608b17

something like....
```
self.admin.assert_icommand('imeta ls -d %s' % (object_name), 'STDOUT_SINGLELINE', ['attribute: ' + attribute])
```